vernac/egramcoq.ml

```diff
@@ -546,15 +546,29 @@ let extend_constr state forpat ng =
 
 let constr_levels = GramState.field ()
 
+let is_disjunctive_pattern_rule ng =
+  String.is_sub "( _ | " (snd ng.notgram_notation) 0
+
+let warn_disj_pattern_notation =
+  let open Pp in
+  let pp ng = str "Use of " ++ Notation.pr_notation ng.notgram_notation ++
+              str " Notation is deprecated as it is inconsistent with pattern syntax" in
+  CWarnings.create ~name:"disj-pattern-notation" ~category:"notation" ~default:CWarnings.Disabled pp
```

Comment:
If there is a new warning, it should be documented (use the `.. warn::` directive). This is true all the more if the warning is disabled by default because users will not learn about it unless they read the changelog.

Reply:
Thanks for the tips - I've tried to use them to improve the `changelog` entry.
Regarding the new warning, I hesitated documenting it more in part because it is currently meant as a developer tool. As I've noted, it will be difficult to merge pattern and term grammars while supporting `(p | q)` notation only for terms, so it's probably best to phase out such notation, and this is what this for. However, it didn't see reasonable to turn it on by default while standard library mainstays use it - all clients of Numbers would get that warning.
  Perhaps it could be activated after switching to some other notation for divisibility in Numbers, which I felt is way beyond the scope of this PR, and somehow disabling it for the compatibility notation that would need to be put in place (or wait until the latter is phased out as well).

Reply:
@Zimmi48: I moved documentation of the warning to the reference manual body, as putting a `..warn` directive in a `changelog` entry triggers a `coq.warn empty` error.

Reply:
Indeed, I may have not been clear but this is what I was suggesting.